.github/workflows/test-pr.yml

```diff
@@ -186,6 +209,8 @@ jobs:
           install_url: https://releases.nixos.org/nix/nix-2.13.3/install
           extra_nix_config: |
             access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+            substituters = http://cache.nixos.org https://cache.iog.io
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
       - name: 'Install Cachix'
```

Comment:
Is this change necessary? Why so?

Reply:
we are implicitly building the haskell/booster backend which relies on the iog cache. best keep it in on the public runner. our self hosted runners (should) have the same cache set up